validator/keymanager/remote-web3signer/internal/client.go

```diff
@@ -87,29 +86,26 @@ func (client *ApiClient) Sign(ctx context.Context, pubKey string, request SignRe
 }
 
 // GetPublicKeys is a wrapper method around the web3signer publickeys api (this may be removed in the future or moved to another location due to its usage).
-func (client *ApiClient) GetPublicKeys(ctx context.Context, url string) ([][fieldparams.BLSPubkeyLength]byte, error) {
-	resp, err := client.doRequest(ctx, http.MethodGet, url, nil /* no body needed on get request */)
+func (client *ApiClient) GetPublicKeys(ctx context.Context, url string) ([]string, error) {
+	resp, err := client.doRequest(ctx, http.MethodGet, url, http.NoBody)
 	if err != nil {
 		return nil, err
 	}
 	var publicKeys []string
 	if err := unmarshalResponse(resp.Body, &publicKeys); err != nil {
 		return nil, err
 	}
-	decodedKeys := make([][fieldparams.BLSPubkeyLength]byte, len(publicKeys))
-	var errorKeyPositions string
-	for i, value := range publicKeys {
-		decodedKey, err := hexutil.Decode(value)
+	if len(publicKeys) != 0 {
+		// check if it's a hex and a public key
+		b, err := hexutil.Decode(publicKeys[0])
```

Comment:
Should we not be checking them all?

Reply:
we'll be checking again and decoding properly after it's used so i was thinking it'd be redundant to do too many

Reply:
So in this case why checking the first key?
I mean, there is no real reason to handle the first key differently than the others right?